docs/docs/35-references/10-promotion-steps.md

```diff
@@ -300,6 +300,53 @@ steps:
 # Render manifests to ./out, commit, push, etc...
 ```
 
+### `delete`
+
+`delete` deletes a file or directory.
+
+#### `delete` Configuration
+
+| Name      | Type | Required | Description                              |
+|-----------|------|----------|------------------------------------------|
+| `path`    | `string` | Y | Path to the file or directory to delete. |
+
+#### `delete` Example
+
+The most common usage of this step is to clean up unwanted files
+or directories before proceeding to subsequent steps in the promotion
+process. For instance, you might use the delete step to remove
+intermediate configurations or artifacts generated in earlier steps.
+
+Consider a scenario where a Stage combines content from two working trees
+to render Stage-specific manifests. If temporary files are generated during
+this process, you can use the delete step to clean them up:
+
+```yaml
+vars:
+- name: gitRepo
+  value: https://github.com/example/repo.git
+steps:
+- uses: git-clone
+  config:
+    repoURL: ${{ vars.gitRepo }}
+    checkout:
+    - commit: ${{ commitFrom(vars.gitRepo, warehouse("base")).ID }}
+      path: ./src
+    - commit: ${{ commitFrom(vars.gitRepo, warehouse(ctx.stage + "-overlay")).ID }}
+      path: ./overlay
+    - branch: stage/${{ ctx.stage }}
+      create: true
+      path: ./out
+- uses: copy
+  config:
+    inPath: ./overlay/stages/${{ ctx.stage }}/kustomization.yaml
+    outPath: ./src/stages/${{ ctx.stage }}/kustomization.yaml
+- uses: delete
+  config:
+    path: ./overlay/stages/${{ ctx.stage }}/temp-file.yaml
+# Render manifests to ./out, commit, push, etc...
+```
```

Comment:
Thought about this and, ironically, the solution is probably to be more hand-wavy about how the file/dir we're deleting came to be:

~~~suggestion
One, common usage of this step is to remove intermediate files
produced by the promotion process prior to a `git-commit` step:

```yaml
vars:
- name: gitRepo
  value: https://github.com/example/repo.git
steps:
- uses: git-clone
  config:
    repoURL: ${{ vars.gitRepo }}
    checkout:
    - commit: ${{ commitFrom(vars.gitRepo) }}
      path: ./src
    - branch: stage/${{ ctx.stage }}
      create: true
      path: ./out

# Steps that produce intermediate files in ./out...

- uses: delete
  config:
    path: ./out/unwanted/file
- uses: git-commit
  config:
    path: ./out
```
~~~

Reply:
LGTM after this.

Reply:
I'm going to take the liberty of committing this suggestion since we're trying really hard to get v1.2 buttoned up.